controllers/awsmachine_controller.go

```diff
@@ -564,6 +568,20 @@ func (r *AWSMachineReconciler) reconcileLBAttachment(machineScope *scope.Machine
 	}
 
 	elbsvc := elb.NewService(clusterScope)
+
+	// In order o prevent sending request to a "not-ready" control plane machines, it is required to remove the machine
+	// from the ELB as soon as the machine gets delete or when the machine is detected or not running
+	// Nb. this is a reinforcement to the ELB health check, which blocks traffic only after a timeout is reached.
+	if !machineScope.AWSMachine.DeletionTimestamp.IsZero() || !machineScope.InstanceIsRunning() {
+		if err := elbsvc.DeregisterInstanceFromAPIServerELB(i); err != nil {
+			r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "FailedDetachControlPlaneELB",
+				"Failed to deregister control plane instance %q from load balancer: %v", i.ID, err)
+			return errors.Wrapf(err, "could not deregister control plane instance %q from load balancer", i.ID)
+		}
```

Comment:
Might be good to add an event in the successful case here as well.

Reply:
Currently there is no success event for register, so I did not add success event for de-register. Should I add a success event in both cases?

Reply:
A success event would be nice, but we can do it in a follow-up PR if you don't want to include it here.

Reply:
event added